include/vikunja/mem/iterator/ConstantIterator.hpp

```diff
@@ -0,0 +1,267 @@
+                {
+                    ConstantIterator cpy = *this;
+                    --index;
+                    return cpy;
+                }
+
+                /**
+                 * @brief Prefix decrement operator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator--(int)
+                {
+                    --index;
+                    return *this;
+                }
+
+                /**
+                 * @brief Add an index to this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const IdxType idx) const
+                {
+                    return ConstantIterator(v, index + idx);
+                }
+
+                /**
+                 * @brief Add a second constant iterator of the same value to this one
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const ConstantIterator& other) const
+                {
+                    assert(v == other.v && "Can't add constant iterators of different values!");
+                    return ConstantIterator(v, index + other.index);
+                }
+
+                /**
+                 * @brief Subtract an index from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator-(const IdxType idx) const
+                {
+                    return ConstantIterator(v, index - idx);
+                }
+
+                /**
+                 * @brief Subtract a second constant iterator of the same value from this one
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator-(const ConstantIterator& other) const
+                {
+                    assert(v == other.v && "Can't subtract constant iterators of different values!");
+                    return ConstantIterator(v, index - other.index);
+                }
+
+                /**
+                 * @brief Add an index to this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator& operator+=(const IdxType idx)
+                {
+                    index += idx;
+                    return *this;
+                }
+
+                /**
+                 * @brief Subtract an index from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator& operator-=(const IdxType idx)
+                {
+                    index -= idx;
+                    return *this;
+                }
+
+#pragma endregion arithmeticoperators
+
+#pragma region comparisonoperators
+
+// if spaceship operator is available is being used we can use spaceship operator magic
+#ifdef USESPACESHIP
+
+                /**
+                 * @brief Spaceship operator for comparisons
+                 */
+                NODISCARD ALPAKA_FN_INLINE auto operator<=>(const ConstantIterator& other) const noexcept = default;
```

Comment:
Such comments don't increase clarity of the code and only add noise. Try to avoid comments for the sake of comments, if they don't add any additional information the code is not already giving.

Reply:
There is also the philosophy to have a doxygen comment for every function, even if it's fairly (or very) obvious what the function does. But I can also remove them if that's not the case for this project. Should I also remove them for the other straight forward arithmetic operators then?

Reply:
To be honest, there is no real guide for the doxygen yet. The student who started this project, did it in some way. I don't believe, that he got some feedback. I want to overwork this in the near future, see #65 . But Bernhard is right, the function is already defined by the concept `iterator`. Therefore the one line which explains the obvious is not necessary.